packages/treat/webpack-plugin/loader.js

```diff
@@ -278,5 +280,31 @@ const serializeTreatModule = (loader, cssRequests, exports) => {
       : `export var ${key} = ${stringifyExports(exports[key])};`,
   );
 
-  return [...sortedCssImports, ...moduleExports].join('\n');
+  const outputCode = [...sortedCssImports, ...moduleExports];
+
+  if (hmr) {
+    outputCode.push(`
+    if (module.hot) {
+      const exportHash = '${loaderUtils.interpolateName(
+        loader,
+        '[contenthash]',
+        {
+          content: moduleExports.join(),
+        },
+      )}';
+            
+      if(module.hot.data && typeof module.hot.data.oldExportHash === 'string' && module.hot.data.oldExportHash !== exportHash) {
```

Comment:
Sorry, but this upsets me.

```suggestion
      if (module.hot.data && typeof module.hot.data.oldExportHash === 'string' && module.hot.data.oldExportHash !== exportHash) {
```